sdk/python/kfp/v2/compiler/pipeline_spec_builder.py

```diff
@@ -357,6 +356,11 @@ def build_importer_spec_for_task(
         type_schema=type_schema,
         reimport=task.importer_spec.reimport)
 
+    if task.importer_spec.metadata and len(task.importer_spec.metadata) != 0:
```

Comment:
nit: `len(task.importer_spec.metadata) != 0` is redundant, the first condition covers it already.